src/array_unique.rs

```diff
@@ -0,0 +1,50 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2022
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::Python;
+
+use hashbrown::HashMap;
+use ndarray::{ArrayView1, Axis};
+use numpy::{IntoPyArray, PyReadonlyArray2};
+
+#[pyfunction]
+#[pyo3(text_signature = "(array, /)")]
+pub fn unique(array: PyReadonlyArray2<u16>, py: Python) -> (PyObject, PyObject) {
```

Comment:
The one thing that I think would be good to add here is a docstring (with three `///` instead of two `//` which is a comment). The rust docstring will be set as `__doc__` on the Python function, so while we won't be including this in the hosted documentation if we document it here IDEs will be able to show the documentation during development using this function.

Reply:
Thanks. I added docstrings with `///`.